### DISCUSSION TITLE
Reconnect

### DISCUSSION THREAD

**Discussion by InterloperX** [FreeOpcUa/opcua-asyncio#473] [Q&A]
Hi all,

Is there an example how to re-establish connection with server and creating new session when the connection to the server is lost?

**Comment by AndreasHeine** [comment]
the official recommendation:
https://reference.opcfoundation.org/v104/Core/docs/Part4/6.7/

**Reply by InterloperX** [reply]
I saw this before, but I am pretty new to python so I was hoping that there is an example that could be used.

**Reply by AndreasHeine** [reply]
unfortunately not^^

**Reply by HarrySky** [reply]
Here is a little simplified snippet from my code. If keepalive() fails with exception - you are disconnected

```python
from asyncua import Client # type: ignore[import]
from asyncua.ua import ObjectIds # type: ignore[import]
from asyncua.ua.uatypes import NodeId # type: ignore[import]

#: Server state node address for keep-alive checks
_SERVER_STATE = NodeId(ObjectIds.Server_ServerStatus_State)

opc_client = Client(URL)
await opc_client.connect()

# Get Server State node (for keepalive)
server_state = opc_client.get_node(_SERVER_STATE)

async def keepalive() -> None:
    await server_state.read_value() 
```

EDIT: And when you know you are disconnected - create new connection 🤷‍♂️

**Reply by InterloperX** [reply]
Thank you very much, I will try to implement it.

**Comment by nkawerau** [comment]
I had the same question. There is an example here in the meantime: https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/client-reconnect.py
